avm/res/network/private-endpoint/main.bicep

```diff
@@ -0,0 +1,192 @@
+metadata name = 'Private Endpoints'
+metadata description = 'This module deploys a Private Endpoint.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Optional. Name of the private endpoint resource to create.')
+param name string = 'pep-${last(split(serviceResourceId, '/'))}'
```

Comment:
Should this not have a default as a resource module?

But when it's used, optionally as an interface, it's should have this default name as it's not the primary resource?

Reply:
I added this today as per the conversation about CAF naming. You tell me @jtracey93 ;) It's now essentially applying the same default as if it were used by another resource with the exception that we also add the `groupId` there.

Reply:
As this is being pushed as a resource module we should align to SNFR25 (https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr25---category-composition---resource-naming)

So technically this should not have a default for the name as it's a resource module. 

But if it's used in a cross reference, say for a key vault 😁, to implement private endpoints. Then in that calla of the module the name can be defaulted to as you show. 

That's what I intended from what I published for SNFR25 above.

@AlexanderSehr

Reply:
Fair, undone.